main.cpp

```diff
@@ -4,64 +4,496 @@ Project 5: Part 1 / 4
+ */
 
-send me a DM to check your pull request
+#include<iostream>
 
- Wait for my code review.
+/*
+ copied UDT 1:====================================
  */
+struct Safeway//2
+{
+    int b;
+    float bb{2.2f};
+    Safeway();
+    ~Safeway();
+    float openTime = 9.f;
+    float discount = 0.7f;
+    int customerPoints = 234;
+    int permissionYear = 2021;
+    int coupons = 34;
+    
+    struct Customer
+    {
+        int accountnumber = 133423;
+        int membership = 37;
+        bool relatedCreditCard = 0;
+        float amountOfLastOrder = 67.59f;
+        int registrationDays = 133;
+        void redeemGiftCard(int codeOfGiftcard = 133278645);
+        bool redeemProduct(int pointsOfProduct = 700);
+        void changeProfile(int dateOfBirthday = 13);
+        int alarmForMembership(int threshold, int currentDays);
+    };
+
+    void orderGroceryOnline(Customer customer);
+    double refund(int productNumber);
+    void collectCoupons(Customer customer);
+    void printThing();
+    int limitOfCoupons(int threshold, int couponsInUse);
+};
+
+Safeway::~Safeway()
+{
+    std::cout<< "Artist is being deleted" << std::endl;
+}
+
+int Safeway::Customer::alarmForMembership(int threshold, int currentDays = 0)//P5
+{  
+    while(currentDays >  threshold)
+    {
+        --currentDays;
+        std::cout << "left days: " << currentDays << std::endl;
+        if(currentDays <= threshold)
+        {
+            std::cout<< "It's time to refresh your membership." << std::endl;
+            membership = currentDays;
+            return currentDays;
+        }
+    }
+    return -1;
+}
+
+int Safeway::limitOfCoupons(int threshold, int couponsInUse = 0)//P5
+{
+    while(couponsInUse < threshold)
+    {
+        ++couponsInUse;
+        std::cout << "coupons in use : " << couponsInUse << std::endl;
+        if(couponsInUse >= threshold)
+        {
+            std::cout << "In one order, you could only use 5 promocode." << std::endl;
+            coupons  = couponsInUse;
+            return couponsInUse;
+        }
+    }
+    return -1;
+}
+
+Safeway::Safeway() : b(2)
+{
+}
+
+
+void Safeway::printThing()
+{
+    std::cout << "Safeway::printThing() b: " << b << " bb:"<< bb << std::endl; 
+}
+
+void Safeway::orderGroceryOnline(Customer customer)
+{
+    std::cout << "Customer's profile: " << customer.accountnumber << std::endl;
+}
+
+double Safeway::refund(int productNumber)
+{
+    std::cout << "Product Number: " << productNumber << std::endl;
+    std::cout << "Product Value: 1" << std::endl;
+    return 0.1;
+}
+
+void Safeway::collectCoupons(Customer customer)
+{
+    std::cout << "Coupons " << customer.membership << std::endl;
+}
 
+void Safeway::Customer::redeemGiftCard(int codeOfGiftcard)
+{
+    std::cout << "submit " << codeOfGiftcard << std::endl;
+}
+
+bool Safeway::Customer::redeemProduct(int pointsOfProduct)
+{
+    if (pointsOfProduct > 700)
+    {
+        return false;
+    }
+    return true;   
```

Comment:
you can simply write `return ! (pointsOfProduct > 700);` or `return pointsOfProduct <= 700;`
- [ ] revise